src/status_im/contexts/wallet/sheets/network_selection/view.cljs

```diff
@@ -0,0 +1,78 @@
+(ns status-im.contexts.wallet.sheets.network-selection.view
+  (:require [quo.core :as quo]
+            [quo.foundations.resources :as quo.resources]
+            [react-native.core :as rn]
+            [status-im.constants :as constants]
+            [status-im.contexts.wallet.common.utils :as utils]
+            [status-im.contexts.wallet.sheets.network-selection.style :as style]
+            [utils.i18n :as i18n]
+            [utils.money :as money]
+            [utils.re-frame :as rf]))
+
+(defn- network-item
+  [{:keys [network token on-select-network currency currency-symbol]}]
+  (let [{:keys [network-name
+                chain-id]} network
+        {:keys [balances-per-chain
+                decimals]} token
+        balance-for-chain  (get balances-per-chain chain-id)
+        total-balance      (money/token->unit (:raw-balance balance-for-chain) decimals)
+        fiat-value         (utils/calculate-token-fiat-value
+                            {:currency currency
+                             :balance  total-balance
+                             :token    token})
+        crypto-formatted   (utils/get-standard-crypto-format token total-balance)
+        fiat-formatted     (utils/get-standard-fiat-format crypto-formatted
+                                                           currency-symbol
+                                                           fiat-value)
+        token-symbol       (:symbol token)
+        mainnet?           (= network-name constants/mainnet-network-name)]
+    [quo/network-list
+     {:label           (name network-name)
+      :network-image   (quo.resources/get-network network-name)
+      :token-value     (str crypto-formatted " " token-symbol)
+      :fiat-value      fiat-formatted
+      :on-press        #(on-select-network network)
+      :container-style (style/network-list-container mainnet?)}]))
```

Comment:
Sorry, I'd be blocking this PR because these calculations are being performed on the view side, they should be calculated from the subscription, we recently had a similar issue with the activity tab. Better to fix them now than later.

Reply:
I really appreciate your review! I will make the necessary changes and let you know. Thank you 😊

Reply:
Thank you @ajayesivan  :)

Reply:
Hi @ulisesmac, I have moved the logic to subscription. Could you please have another look